roles/jd-client/src/lib/job_declarator/mod.rs

```diff
@@ -87,23 +87,23 @@ impl JobDeclarator {
         let stream = tokio::net::TcpStream::connect(address).await?;
         let initiator = Initiator::from_raw_k(authority_public_key)?;
         let (mut receiver, mut sender, _, _) =
-            Connection::new(stream, HandshakeRole::Initiator(initiator))
-                .await
-                .expect("impossible to connect");
+            Connection::new(stream, HandshakeRole::Initiator(initiator)).await?;
 
-        let proxy_address = SocketAddr::new(
-            IpAddr::from_str(&config.downstream_address).unwrap(),
-            config.downstream_port,
-        );
+        let downstream_address = IpAddr::from_str(&config.downstream_address)?;
+        let proxy_address = SocketAddr::new(downstream_address, config.downstream_port);
 
         info!(
             "JD proxy: setupconnection Proxy address: {:?}",
             proxy_address
         );
 
-        SetupConnectionHandler::setup(&mut receiver, &mut sender, proxy_address)
-            .await
-            .unwrap();
+        match SetupConnectionHandler::setup(&mut receiver, &mut sender, proxy_address).await {
+            Ok(_) => {}
+            Err(e) => {
+                error!("Error setting up connection: {:?}", e);
+                return Err(Error::ConnectionSetup);
+            }
+        };
```

Comment:
Since the `Ok` case doesn't do anything, using `if let` is cleaner and makes the code easier to read.
```suggestion
   if let Err(e) = SetupConnectionHandler::setup(&mut receiver, &mut sender, proxy_address).await {
    error!("Error setting up connection: {:?}", e);
    return Err(Error::ConnectionSetup);
}
```

Reply:
Ah, good point. I left the `match` as is and moved the `success` log into the `ok` arm. thank you